ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -200,30 +201,31 @@ private class PostgresLedgerDao(
             nodeId -> party.map(p => Ref.Party.assertFromString(p))
         }
 
-      def acsLookupContract(acs: Unit, cid: AbsoluteContractId) =
-        lookupActiveContractSync(cid).map(_.toActiveContract)
-      //TODO: Implement check whether the given contract key exists
-      def acsKeyExists(acc: Unit, key: GlobalKey): Boolean = false
-      //TODO: store contract key
-      def acsAddContract(
-          acs: Unit,
-          cid: AbsoluteContractId,
-          c: ActiveContracts.ActiveContract,
-          keyO: Option[GlobalKey]): Unit =
-        storeContract(offset, Contract.fromActiveContract(cid, c))
-      //TODO: remove contract key
-      def acsRemoveContract(acs: Unit, cid: AbsoluteContractId, keyO: Option[GlobalKey]): Unit = {
-        archiveContract(offset, cid)
-        ()
+      final class AcsStoreAcc extends ActiveContractsSteps[AcsStoreAcc] {
+
+        def lookupContract(cid: AbsoluteContractId) =
+          lookupActiveContractSync(cid).map(_.toActiveContract)
+        //TODO: Implement check whether the given contract key exists
+        def keyExists(key: GlobalKey): Boolean = false
+        //TODO: store contract key
+        def addContract(
+            cid: AbsoluteContractId,
+            c: ActiveContracts.ActiveContract,
+            keyO: Option[GlobalKey]) = {
+          storeContract(offset, Contract.fromActiveContract(cid, c))
+          this
+        }
+        //TODO: remove contract key
+        def removeContract(cid: AbsoluteContractId, keyO: Option[GlobalKey]) = {
+          archiveContract(offset, cid)
+          this
+        }
+        def implicitlyDisclose(global: Relation[AbsoluteContractId, Ref.Party]) =
+          this // TODO SC
```

Comment:
we'll need an additional table here -- let's not do this in this pr.

Reply:
5dd7f557d90d82b19f68e09ad9d87dbb0b8082f6